packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsTeleporterSteps.tsx

```diff
@@ -0,0 +1,182 @@
+import { ExternalLink } from '../common/ExternalLink'
+import { Step } from './TransactionsTableDetailsSteps'
+import { DepositCountdown } from '../common/DepositCountdown'
+
+const TeleportMiddleStepFailureExplanationNote = ({
+  tx
+}: {
+  tx: MergedTransaction
+}) => {
+  return (
+    <div className="mt-2">
+      <div
+        className={twMerge(
+          'flex items-center gap-2',
+          l1L2RetryableRequiresRedeem(tx) && 'text-white/60'
+        )}
+      >
+        {!l1L2RetryableRequiresRedeem(tx) ? (
+          <CheckCircleIcon
+            height={15}
+            className="h-[15px] w-[15px] shrink-0 text-green-400"
+          />
+        ) : (
+          <div className="m-[1px] h-[13px] w-[13px] shrink-0 rounded-full border border-white/60" />
+        )}
+        Funds deposited to {getNetworkName(getChainIdForRedeemingRetryable(tx))}
+        .{' '}
+      </div>
+      <div
+        className={twMerge(
+          'mt-1 flex items-center gap-2',
+          l2ForwarderRetryableRequiresRedeem(tx) && 'text-white/60'
+        )}
+      >
+        {!l2ForwarderRetryableRequiresRedeem(tx) ? (
+          <CheckCircleIcon
+            height={15}
+            className="h-[15px] w-[15px] shrink-0 text-green-400"
+          />
+        ) : (
+          <div className="m-[1px] h-[13px] w-[13px] shrink-0 rounded-full border border-white/60" />
+        )}
+        Funds sent to {getNetworkName(tx.destinationChainId)}.
+      </div>
+    </div>
+  )
+}
+
+export const TransactionsTableDetailsTeleporterSteps = ({
+  tx,
+  address
+}: {
+  tx: MergedTransaction
+  address: Address | undefined
+}) => {
+  const { isTestnet: isTestnetTx } = isNetwork(tx.childChainId)
+
+  const l2TxID = tx.l1ToL2MsgData?.l2TxID
+  const isFirstRetryableLegSucceeded =
+    typeof l2TxID !== 'undefined' &&
+    typeof tx.l2ToL3MsgData?.l2ForwarderRetryableTxID === 'undefined'
+  const l2ChainId = tx.l2ToL3MsgData?.l2ChainId
+  const isFirstRetryableLegFailed = firstRetryableLegRequiresRedeem(tx)
+  const l2ForwarderRequiresRedeem = l2ForwarderRetryableRequiresRedeem(tx)
+
+  const isFirstRetryableLegResolved =
+    isFirstRetryableLegSucceeded || isFirstRetryableLegFailed
+
+  const isSecondRetryableLegResolved =
+    secondRetryableLegForTeleportRequiresRedeem(tx) ||
+    typeof tx.l2ToL3MsgData?.l3TxID !== 'undefined'
+
+  const firstTransactionText = useMemo(() => {
+    if (isFirstRetryableLegFailed) {
+      const l2NetworkName = getNetworkName(getChainIdForRedeemingRetryable(tx))
+
+      return (
+        <div>
+          Transaction failed on {l2NetworkName}. You have 7 days to try again.
+          After that, your funds will be{' '}
+          <span className="font-bold text-red-400">lost forever</span>.
+          {/* if we detect we will have 2 redemptions in the first leg of teleport, explain it to users */}
+          {l2ForwarderRequiresRedeem && (
+            <TeleportMiddleStepFailureExplanationNote tx={tx} />
+          )}
+        </div>
```

Comment:
Abstracted this `Transaction failed on {l2NetworkName}. You have 7 days to try again.
          After that, your funds will be{' '}
          <span className="font-bold text-red-400">lost forever</span>.`, since it is being used in 2 places.